indexing-service/src/main/java/org/apache/druid/indexing/common/TaskToolbox.java

```diff
@@ -98,35 +98,36 @@
   private final DruidNode druidNode;
   private final LookupNodeService lookupNodeService;
   private final DataNodeService dataNodeService;
+  private final boolean useOak;
```

Comment:
a boolean is fine, but maybe a more generic option is to use enum that can be extended in the future

Reply:
I agree that an enum may be better than a boolean here. 

I was hoping to get some community feedback first whether this kind of configuration is indeed a viable option, or if they would suggest doing it a different way completely.